WordPress/Sniffs/NamingConventions/PrefixAllGlobalsSniff.php

```diff
@@ -0,0 +1,503 @@
+			if ( false === $array_open ) {
+				// Live coding.
+				return;
+			}
+
+			$array_key = $this->phpcsFile->findNext( PHP_CodeSniffer_Tokens::$emptyTokens, ( $array_open + 1 ), null, true, null, true );
+			if ( false === $array_key || T_CONSTANT_ENCAPSED_STRING !== $this->tokens[ $array_key ]['code'] ) {
+				// Key is not a string, nothing to do.
+				return;
+			}
+
+			$stackPtr      = $array_key;
+			$variable_name = $this->strip_quotes( $this->tokens[ $array_key ]['content'] );
+
+		} else {
+			// Local variables in a function do not need to be prefixed unless they are being imported.
+			if ( $this->phpcsFile->hasCondition( $stackPtr, array( T_FUNCTION, T_CLOSURE ) ) === true ) {
+				$condition = $this->phpcsFile->getCondition( $stackPtr, T_FUNCTION );
+				if ( false === $condition ) {
+					$condition = $this->phpcsFile->getCondition( $stackPtr, T_CLOSURE );
+				}
+
+				$has_global = $this->phpcsFile->findPrevious( T_GLOBAL, ( $stackPtr - 1 ), $this->tokens[ $condition ]['scope_opener'] );
+				if ( false === $has_global ) {
+					// No variable import happening.
+					return;
+				}
+
+				// Ok, this may be an imported global variable.
+				$end_of_statement = $this->phpcsFile->findNext( T_SEMICOLON, ( $has_global + 1 ) );
+				if ( false === $end_of_statement ) {
+					// No semi-colon - live coding.
+					return;
+				}
+
+				for ( $ptr = ( $has_global + 1 ); $ptr <= $end_of_statement; $ptr++ ) {
+					$imported = $this->phpcsFile->findNext( T_VARIABLE, $ptr, $end_of_statement, false, null, true );
+
+					if ( false === $imported ) {
+						// Reached the end of the global statement without finding the variable,
+						// so this must be a local variable.
+						return;
+					}
+
+					if ( substr( $this->tokens[ $imported ]['content'], 1 ) === $variable_name ) {
+						break;
+					}
+
+					// Move the stack pointer forward for the next loop.
+					$ptr = $imported;
+				}
+
+				unset( $condition, $has_global, $end_of_statement, $ptr, $imported );
+
+			} // End if().
+
+			// Properties in a class do not need to be prefixed.
+			$conditions = array_keys( $this->tokens[ $stackPtr ]['conditions'] );
+			$ptr        = array_pop( $conditions );
+			if ( isset( $this->tokens[ $ptr ] )
+				&& in_array( $this->tokens[ $ptr ]['code'], array( T_CLASS, T_ANON_CLASS, T_TRAIT ), true )
+			) {
+				return;
+			}
+
+			// Function parameters do not need to be prefixed.
+			if ( isset( $this->tokens[ $stackPtr ]['nested_parenthesis'] ) ) {
+				foreach ( $this->tokens[ $stackPtr ]['nested_parenthesis'] as $opener => $closer ) {
+					if ( isset( $this->tokens[ $opener ]['parenthesis_owner'] ) && T_FUNCTION === $this->tokens[ $this->tokens[ $opener ]['parenthesis_owner'] ]['code'] ) {
+						return;
+					}
+				}
+			}
+		} // End if().
+
+		// Ignore superglobals and WP global variables.
+		if ( isset( $this->superglobals[ $variable_name ] ) || isset( $this->wp_globals[ $variable_name ] ) ) {
```

Comment:
This condition can be moved up higher in the function (but may need to be repeated?), which would improve performance by bypassing above logic that in this case is unnecessary. The `$variable_name` is not modified in the above `else`, only in the `if`. So probably the best way to do this would be to split the `if and `else` and place this condition between them.